hub/core/chunk_engine.py

```diff
@@ -109,18 +109,47 @@ def __init__(
         self._meta_cache = meta_cache
         self.version_state = version_state
         self.compression = None
-        self._last_row = 0
         self.chunk_class = BaseChunk
 
-        if self.tensor_meta.sample_compression:
-            self.compression = self.tensor_meta.sample_compression
+        self._tensor_meta: Optional[TensorMeta] = None
```

Comment:
this prevents lookup from LRUCache everytime we need tensor meta (also acts as groundwork for replacing cachable in a future PR)